projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1606,6 +1613,105 @@ public function get_social_opengraph_image( $post_id ) {
 		return array();
 	}
 
+	/**
+	 * Returns the image size in bytes of a remote image.
+	 *
+	 * @param  string $image_url       Image URL.
+	 * @return integer|null $bytes      Image size in bytes, or null if request failed.
+	 */
+	public function get_remote_filesize( $image_url ) {
+		$response = wp_remote_get( $image_url, array( 'method' => 'HEAD' ) );
+		$size     = wp_remote_retrieve_header( $response, 'content-length' );
+
+		return ! empty( $size ) ? $size : null;
+	}
+
+	/**
+	 * Returns the resized Photon URL for a given image.
+	 *
+	 * @param string $image_url Image URL.
+	 * @param int    $width Image width.
+	 * @param int    $height Image height.
+	 * @return string
+	 */
+	public function get_resized_image_url( $image_url, $width, $height ) {
+		return jetpack_photon_url(
+			$image_url,
+			array(
+				'w' => $width,
+				'h' => $height,
+			)
+		);
+	}
+
+	/**
+	 * This function runs the image through Proton to compress it, and also scales it down if needed.
+	 *
+	 * @param string $url Image URL.
+	 * @param int    $width Image width.
+	 * @param int    $height Image height.
+	 * @return array The compressed image data.
+	 */
+	public function compress_and_scale_og_image( $url, $width, $height ) {
+		// If the dimensions are fine we just run it through Proton to compress it.
+		if ( 1200 >= $width && 1200 >= $height ) {
```

Comment:
Should this be either dimension? If an image is 5000x1000, then perhaps we should resize it?

Reply:
I think that's how it works here, it's just the negated decision. If both are smaller than we do not scale it, just run it through for the compression. Then if either is bigger, we do the same with the new height and width below

Reply:
Oh I read it wrong 🤦